drivers/npmx/npmx_shell.c

```diff
@@ -4391,61 +4388,6 @@ SHELL_STATIC_SUBCMD_SET_CREATE(sub_leds, SHELL_CMD(mode, &sub_leds_mode, "LEDs m
 			       SHELL_CMD(state, &sub_leds_state, "LEDs state", NULL),
 			       SHELL_SUBCMD_SET_END);
 
-/* Creating subcommands (level 3 command) array for command "gpio status". */
-SHELL_STATIC_SUBCMD_SET_CREATE(sub_gpio_status,
-			       SHELL_CMD(get, NULL, "Get GPIO status", cmd_gpio_status_get),
-			       SHELL_SUBCMD_SET_END);
-
-/* Creating subcommands (level 3 command) array for command "gpio mode". */
-SHELL_STATIC_SUBCMD_SET_CREATE(
-	sub_gpio_mode, SHELL_CMD(get, NULL, "Get GPIO mode configuration", cmd_gpio_mode_get),
-	SHELL_CMD(set, NULL, "Set GPIO mode configuration", cmd_gpio_mode_set),
-	SHELL_SUBCMD_SET_END);
-
-/* Creating subcommands (level 3 command) array for command "gpio type". */
-SHELL_STATIC_SUBCMD_SET_CREATE(sub_gpio_type,
-			       SHELL_CMD(get, NULL, "Check GPIO type", cmd_gpio_type_get),
-			       SHELL_SUBCMD_SET_END);
-
-/* Creating subcommands (level 3 command) array for command "gpio pull". */
-SHELL_STATIC_SUBCMD_SET_CREATE(sub_gpio_pull,
-			       SHELL_CMD(get, NULL, "Get pull configuration", cmd_gpio_pull_get),
-			       SHELL_CMD(set, NULL, "Set pull configuration", cmd_gpio_pull_set),
-			       SHELL_SUBCMD_SET_END);
-
-/* Creating subcommands (level 3 command) array for command "gpio drive". */
-SHELL_STATIC_SUBCMD_SET_CREATE(
-	sub_gpio_drive, SHELL_CMD(get, NULL, "Get drive current configuration", cmd_gpio_drive_get),
-	SHELL_CMD(set, NULL, "Set drive current configuration", cmd_gpio_drive_set),
-	SHELL_SUBCMD_SET_END);
-
-/* Creating subcommands (level 3 command) array for command "gpio open_drain". */
-SHELL_STATIC_SUBCMD_SET_CREATE(sub_gpio_open_drain,
-			       SHELL_CMD(get, NULL, "Check if open drain is enabled",
-					 cmd_gpio_open_drain_get),
-			       SHELL_CMD(set, NULL, "Enable or disable open drain",
-					 cmd_gpio_open_drain_set),
-			       SHELL_SUBCMD_SET_END);
-
-/* Creating subcommands (level 3 command) array for command "gpio debounce". */
-SHELL_STATIC_SUBCMD_SET_CREATE(sub_gpio_debounce,
-			       SHELL_CMD(get, NULL, "Check if debounce is enabled",
-					 cmd_gpio_debounce_get),
-			       SHELL_CMD(set, NULL, "Enable or disable debounce",
-					 cmd_gpio_debounce_set),
-			       SHELL_SUBCMD_SET_END);
-
-/* Creating subcommands (level 2 command) array for command "gpio". */
-SHELL_STATIC_SUBCMD_SET_CREATE(
-	sub_gpio, SHELL_CMD(status, &sub_gpio_status, "GPIO status", NULL),
-	SHELL_CMD(mode, &sub_gpio_mode, "GPIO mode configuration", NULL),
-	SHELL_CMD(type, &sub_gpio_type, "GPIO type", NULL),
```

Comment:
Why was this command removed?

Reply:
Because in the `npmx_gpio_config_t` struct there is no `type` field.

```
/** @brief Configuration structure for GPIO. */
typedef struct
{
    npmx_gpio_mode_t  mode;       ///< Configuration for GPIO mode selection.
    npmx_gpio_drive_t drive;      ///< GPIO drive current.
    npmx_gpio_pull_t  pull;       ///< GPIO pull mode.
    bool              open_drain; ///< Enable GPIO open drain.
    bool              debounce;   ///< Enable GPIO debounce.
} npmx_gpio_config_t;
```

Reply:
It was a helper function to check if it's output or input:
https://github.com/NordicSemiconductor/npmx-zephyr/pull/73/commits/ccb82da71c9a6404d90bdf60313ca1a205e70519#diff-0e688e7fadaa6f2edadba6ab36e3e5be7963e487870b190f00eeceda2c0fc626L3015-L3020
I think it should stay.

Reply:
Right, reverted.